packages/jest/src/generators/init/init.ts

```diff
@@ -34,74 +35,89 @@ const schemaDefaults = {
   rootProject: false,
 } as const;
 
+const FILE_EXTENSION_REGEX = /(?<!(^|\/))(\.[^/.]+)$/;
+
+function generateGlobalConfig(tree: Tree, isJS: boolean) {
+  const contents = isJS
+    ? stripIndents`
+    const { getJestProjects } = require('@nrwl/jest');
+
+    module.exports = {
+      projects: getJestProjects()
+    };`
+    : stripIndents`
+    import { getJestProjects } from '@nrwl/jest';
+
+    export default {
+     projects: getJestProjects()
+    };`;
+  tree.write(`jest.config.${isJS ? 'js' : 'ts'}`, contents);
+}
+
 function createJestConfig(tree: Tree, options: NormalizedSchema) {
   if (!tree.exists('jest.preset.js')) {
     // preset is always js file.
     tree.write(
       `jest.preset.js`,
       `
       const nxPreset = require('@nrwl/jest/preset').default;
-     
+
       module.exports = { ...nxPreset }`
     );
 
     addTestInputs(tree);
   }
-  const rootJestConfig = findRootJestConfig(tree);
-
-  if (options.rootProject && !rootJestConfig) {
-    // we don't want any config to be made because the jest-project generator
-    // will make the config when using rootProject
+  if (options.rootProject) {
+    // we don't want any config to be made because the `jestProjectGenerator`
+    // will copy the template config file
+    return;
+  }
+  const rootJestPath = findRootJestConfig(tree);
+  if (!rootJestPath) {
+    // if there's not root jest config, we will create one and return
+    // this can happen when:
+    // - root jest config was renamed => in which case there is migration needed
+    // - root project didn't have jest setup => again, no migration is needed
+    generateGlobalConfig(tree, options.js);
     return;
   }
+
   const isProjectConfig =
-    rootJestConfig &&
-    tree.exists(rootJestConfig) &&
-    !tree.read(rootJestConfig, 'utf-8').includes('getJestProjects()');
-  if (!options.rootProject && isProjectConfig) {
-    // moving from single project to multi project.
-    // TODO(caleb): this is brittle and needs to be detected better?
-    const nxJson = readJson(tree, 'nx.json');
```

Comment:
nx.json might not have `defaultProject` set, but root project's configuration must have `root: "."`